cleanrepo/CleanRepo/Program.cs

```diff
@@ -1591,6 +1718,67 @@ private static List<FileInfo> GetYAMLFiles(string directoryPath)
 
         return mediaFiles;
     }
+    /// <summary>
+    /// Returns a dictionary of all .png/.jpg/.gif/.svg files in the directory.
+    /// The search includes the text found in the files.
+    /// </summary>
+    private static Dictionary<string, string> ScanMediaFiles(List<string>? imageFilePaths, string ocrModelDirectory)
+    {
+
+        Dictionary<string, string> ocrDataForFiles = new(StringComparer.InvariantCultureIgnoreCase);
+
+        if (imageFilePaths is null or { Count : 0 })
+        {
+            Console.WriteLine("\nNo .png/.jpg/.gif/.svg files to scan!");
+            return ocrDataForFiles;
+        }
+
+        using (var engine = new TesseractEngine(ocrModelDirectory, "eng", EngineMode.Default))
+        {
+            foreach (string imageFilePath in imageFilePaths)
+            {
+                using (var img = Pix.LoadFromFile(imageFilePath))
+                {
+                    using (Page page = engine.Process(img))
+                    {
+                        string text = page.GetText();
+                        ocrDataForFiles.Add(imageFilePath, text);
+                    }
+                }
+            }
+        }
+        return ocrDataForFiles;
```

Comment:
This approach uses a concurrent dictionary and then fans out in parallel for the execution of each image. This should make it much faster, and based on my reading of `TesseractEngine` they fully support parallelization like this.

```suggestion
        ConcurrentDictionary<string, string> ocrDataForFiles = new(StringComparer.InvariantCultureIgnoreCase);
        
        if (imageFilePaths is null or { Count: 0 })
        {
            Console.WriteLine("\nNo .png/.jpg/.gif/.svg files to scan!");
            return ocrDataForFiles;
        }
        
        using var engine = new TesseractEngine(ocrModelDirectory, "eng", EngineMode.Default);
        
        Parallel.ForEach(imageFilePaths, imageFilePath =>
        {
            using var img = Pix.LoadFromFile(imageFilePath);
            using Page page = engine.Process(img);
        
            string text = page.GetText();
        
            ocrDataForFiles.AddOrUpdate(
                imageFilePath, _ => text, (_, __) => text);
        });
        
        return ocrDataForFiles;
```